libs/rtefsutils/test/src/RteFsUtilsTest.cpp

```diff
@@ -727,33 +727,43 @@ TEST_F(RteFsUtilsTest, RemoveDirectoryAutoRetry) {
 TEST_F(RteFsUtilsTest, SetFileReadOnly) {
   bool ret;
   error_code ec;
-  const fs::perms write_mask = fs::perms::owner_write | fs::perms::group_write | fs::perms::others_write;
+  constexpr fs::perms write_mask = fs::perms::owner_write | fs::perms::group_write | fs::perms::others_write;
+  constexpr fs::perms exec_mask = fs::perms::owner_exec | fs::perms::group_exec | fs::perms::others_exec;
 
   RteFsUtils::CreateFile(filenameRegular, "foo");
 
+  // Set permission to a known value (0644), with out this, the tests below will fail.
+  fs::permissions(filenameRegular, fs::perms::owner_write | fs::perms::owner_read | fs::perms::group_read | fs::perms::others_read, ec);
+
   // Test filename with regular separators
   ret = RteFsUtils::SetFileReadOnly(filenameRegular, true);
   EXPECT_EQ(ret, true);
   EXPECT_EQ((fs::status(filenameRegular, ec).permissions() & write_mask), fs::perms::none);
+  EXPECT_EQ(fs::status(filenameRegular, ec).permissions() & exec_mask, fs::perms::none);
```

Comment:
I've tried to figure out why this is not working on Windows and it appears that it might not be so strange after all.
Based on your comments @JonatanAntoni, I added the verification that the execute flag was removed, but this test will never work on Windows. See the [Mircorsoft STL implementation of `_Permissions`](https://github.com/microsoft/STL/blob/main/stl/inc/filesystem#L3974) for the reason.

Is it okay to just drop the check of on the exec bits or should I make the test conditional using ifdef?

Reply:
This test makes me wondering. I think we agreed that we won't touch any flags other than the write ones. We should never remove exec flags nor set them. Hence the check should be removed, anyway.

Reply:
Ah, sorry, I misunderstood your comment. You wanted to add a check that only the write flags got affected and no others. Fair point.

Instead of check the flags explicitly; would it work to store the current permission of the file right after creation? And then verify after `SetFileReadOnly` only the write flags are changed/removed? The status of all other flags should just be untouched.